### ISSUE TITLE
[BUG] Button "Take Snapshot" and "Create Backup" grayed out.

### ISSUE THREAD

**Issue by chriscchien** [longhorn/longhorn#6841]
## Describe the bug (🐛 if you encounter this issue)

In longhorn v1.6.0-dev UI, on the volume detail page, button "Take Snapshot" and "Create Backup" grayed out even when the volume is attached and backup target is setup correctly. (Create backup through API is still working(test [result](https://ci.longhorn.io/job/private/job/longhorn-tests-regression/4991/testReport/tests/test_basic/))) 

![image](https://github.com/longhorn/longhorn/assets/73337386/6cc3aee8-70f4-4c5a-b689-198e29e67704)

## To Reproduce

1. Deploy longhorn v1.6.0-dev
2. Setup backup target
3. Create volume and attach to a node
4. Navigate to volume detail page
5. Button "Take Snapshot" and "Create Backup" grayed out.

## Expected behavior

Button "Take Snapshot" should in enable state when volume attached
Button "Create Backup" should in enable state when volume attached and when backup target setup correcty

## Support bundle for troubleshooting

N/A

## Environment

 - Longhorn version: v1.6.0-dev
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): kubectl
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: k3s

## Additional context

Did not see in v1.5 dev

**Comment by PhanLe1010** [comment]
Note that upgrading engine image is also grayed out even though there is a qualified engine image

**Comment by james-munson** [comment]
In addition, the option for "upgrade engine" is absent from the right-hand dropdown menu on the volume itself.

**Comment by james-munson** [comment]
On the other hand, in the view of engine images, the "delete" option is not grayed out, even for an image that cannot be deleted because it is the default image or is actively in use by some volume.

**Comment by james-munson** [comment]
Looking at the conditions for when snapshot create is disabled, I wonder whether something has changed to fool the logic that tests for an engine upgrade in progress, so it always appears to be true.

**Comment by ejweber** [comment]
I did not do any testing on this or follow through on it at all, however, when we were discussing this offline it jogged my memory and I think this might be a good place to start.

Among other conditions, we grey out the `Take Snapshot` button if `volume.currentImage != volume.engineImage`.

https://github.com/longhorn/longhorn-ui/blob/1b2b2db0ad302fd50dc2aec75f16438577a4bd6e/src/routes/volume/helper/index.js#L566-L568

However, we deprecated the `engineImage` field in https://github.com/longhorn/longhorn-manager/pull/2139/files.

There is already a UI issue tracking this at https://github.com/longhorn/longhorn/issues/6685 with an additional comment about engine upgrades not working (similar to @PhanLe1010's above).

Probably a proper implementation of https://github.com/longhorn/longhorn/issues/6685 will resolve this bug.
